### ISSUE TITLE
Neither directory contains a build file meson.build

### ISSUE THREAD

**Issue by nobicycle** [kiwix/kiwix-tools#158]
Thank you for kiwix-tools.
I am here due to these errors from Arch Linuix AUR repo package being unable to find the kiwix binary (actually at /usr/bin/kiwix)

> 
> sudo -u kiwix /usr/lib/kiwix/bin/kiwix-install --verbose --buildIndex addcontent /opt/kiwix/data/content/wikibooks_en_all_nopic_2016-08.zim /usr/bin/kiwix
> Check if the ZIM file exists...
> Check if the ZIM file is valid...
> Check if the Kiwix path is valid...
> Unable to find the Kiwix binary at '/usr/bin/kiwix/kiwix/kiwix[.exe]'.

So I thought I would compile from source as per the guide:
Then install Meson itself:

```
virtualenv -p python3 ./ # Create virtualenv
source bin/activate      # Activate the virtualenv
pip install meson        # Install Meson
hash -r                  # Refresh bash paths

This part is OK.

```
Finally download and build Ninja locally:

```
git clone git://github.com/ninja-build/ninja.git
cd ninja
git checkout release
./configure.py --bootstrap
mkdir ../bin                      
cp ninja ../bin
cd ..

This part is OK
```

Compilation
Once all dependencies are installed, you can compile kiwix-lib with:

```
mkdir build
meson . build
```
This generates an error

**Error during basic setup:

Neither directory contains a build file meson.build.**

Further info:
```
$ pwd
/home/archives/Education/Kiwix/Kiwix-Tools
(Kiwix-Tools) [me@linux Kiwix-Tools]$ ls
bin  build  include  lib  ninja  pip-selfcheck.json  share

```

**Comment by kelson42** [comment]
@nobicycle What do you want to achieve exactly? I suspect `kiwix-install` is not the tool you need.

**Comment by nobicycle** [comment]
Hi Kelson42

I want to configure kiwix-server with some existing zim files. kiwix-install appears to be the correct tool to install zims.

On Linux Arch, the package kiwix-bin provides the following executables;

> kiwix        kiwix-serve     kiwix-install  kiwix-read    kiwix-serve
> kiwix-index  kiwix-manage   kiwix-search  xapian-compact
> 
Regarding the error "Unable to find the Kiwix binary at '/usr/bin/kiwix/kiwix/kiwix[.exe]'", if I fool kiwix-install using a link to kiwix then it complains about not finding xapian.

In summary, I want to run kiwix-server (OK on that) with existing zims.

**Comment by mgautierfr** [comment]
`meson . build` is the step equivalent to the configure step in the autotools system.
It tells meson to build the source in the current directory (`.`) in the build directory (`build`).
So you have to be in the source directory to run this command. Or change the command to `meson <path_to_source_directory> <path_to_build_directory>`

However, I agree with @kelson42, I suppose it is not the tool you need. `kiwix-install` is a pretty old tool that try to locate the desktop version of kiwix. 
`kiwix-serve` can read zim files directly given on it command line or the ones listed in a "library" file (usually `library.xml`). To add zim files to a library file, you have to use `kiwix-manage`.

**Comment by nobicycle** [comment]
Thank you mgautierfr. 
	
I have an account for kiwix. The library is stored here:
	
```
 ls -l  /opt/kiwix/data/library/library.xml 
-rw-r--r-- 1 kiwix kiwix 53 Feb 26 23:13 /opt/kiwix/data/library/library.xml
```
	
I tried kiwix-manage from the Arch Linux AUR repo:
	
```
sudo -u kiwix /bin/bash -c "cd /opt/kiwix; /usr/lib/kiwix/bin/kiwix-manage /opt/kiwix/data/library/library.xml add /media/esata/MEDIA/System/Kiwix/data/content --zimPathToSave='wikipedia_en_for-schools_2013-01.zim'" 
error reading zim-file header
Unable to build or save library file '/opt/kiwix/data/library/library.xml'

```	
Note kiwix does have access to the zims:

```
sudo -u kiwix ls  /media/esata/MEDIA/System/Kiwix/data/content
3d9bd8e7de9e5d25bb1b8a3160839eb7e1eaaada.meta4	da39a3ee5e6b4b0d3255bfef95601890afd80709.meta4	vikidia_en_all_2017-12.zim	      wikipedia_en_wp1-0.8_2016-08.zim.aria2
4550ca1bfd7c5d7eb8f0c56fd0bc47c226285f58.meta4	f2d683dca1b02a7a3043d8563e56e0e5e26e5256.meta4	wikibooks_en_all_2016-08.zim	      wikiquote_id_all_2016-08.zim
4f7808870718a3344fa4401f1a3816c83af8b10e.meta4	index.idx					wikibooks_en_all_nopic_2016-08.zim    wikivoyage_en_all_nopic_2016-08.zim
864c8da01d5b221af8886d9c3654748ca0585715.meta4	phet_en_2017-08.zim				wikipedia_en_for-schools_2013-01.zim
bf2fa7f54fda57428cada41a8cb1b49d0a0f9d61.meta4	phet_en_2017-12.zim				wikipedia_en_medicine.zim
c37effe819d7f594c44d835a933339538228cdf4.meta4	phet_en.zim					wikipedia_en_wp1-0.8_2016-08.zim

```	

I tried variations on the above
	
```
$ sudo -u kiwix /bin/bash -c "cd /opt/kiwix; /home/archives/Education/Kiwix/ftp.fau.de-kiwin-nightly/1/kiwix-manage /opt/kiwix/data/library/library.xml add /media/esata/MEDIA/System/Kiwix/data/content --zimPathToSave='/media/esata/MEDIA/System/Kiwix/data/content/wikipedia_en_for-schools_2013-01.zim'" 
/bin/bash: line 1: 12786 Segmentation fault      (core dumped) /home/archives/Education/Kiwix/ftp.fau.de-kiwin-nightly/1/kiwix-manage /opt/kiwix/data/library/library.xml add /media/esata/MEDIA/System/Kiwix/data/content --zimPathToSave='/media/esata/MEDIA/System/Kiwix/data/content/wikipedia_en_for-schools_2013-01.zim'
	
[me@linux 1]$ sudo -u kiwix /bin/bash -c "cd /opt/kiwix; /home/archives/Education/Kiwix/ftp.fau.de-kiwin-nightly/1/kiwix-manage /opt/kiwix/data/library/library.xml add  --zimPathToSave='/media/esata/MEDIA/System/Kiwix/data/content/wikipedia_en_for-schools_2013-01.zim'" 
error 2 opening file "--zimPathToSave=/media/esata/MEDIA/System/Kiwix/data/content/wikipedia_en_for-schools_2013-01.zim
Unable to build or save library file '/opt/kiwix/data/library/library.xml'

**Comment by mgautierfr** [comment]
You should use this kind of command :
`sudo -u kiwix /bin/bash -c "cd /opt/kiwix; /usr/lib/kiwix/bin/kiwix-manage /opt/kiwix/data/library/library.xml add /media/esata/MEDIA/System/Kiwix/data/content/wikipedia_en_for-schools_2013-01.zim"`

The option `--zimPathToSave` should be used only if you want to store in the library a path different of the real path.

> $ sudo -u kiwix /bin/bash -c "cd /opt/kiwix; /home/archives/Education/Kiwix/ftp.fau.de-kiwin-nightly/1/kiwix-manage /opt/kiwix/data/library/library.xml add /media/esata/MEDIA/System/Kiwix/data/content --zimPathToSave='/media/esata/MEDIA/System/Kiwix/data/content/wikipedia_en_for-schools_2013-01.zim'" 
/bin/bash: line 1: 12786 Segmentation fault      (core dumped) /home/archives/Education/Kiwix/ftp.fau.de-kiwin-nightly/1/kiwix-manage /opt/kiwix/data/library/library.xml add /media/esata/MEDIA/System/Kiwix/data/content --zimPathToSave='/media/esata/MEDIA/System/Kiwix/data/content/wikipedia_en_for-schools_2013-01.zim'

This should not happen. Can you create a new issue for this bug and provide the associated core dump ?

**Comment by kelson42** [comment]
@nobicycle I think the easiest solution is just to start kiwix-serve with a list of path to ZIM files like this:
`kiwix-serve --port=8080 ./foo.zim ./bar.zim`

Try with one of our last nightly https://download.kiwix.org/nightly/ (choose a kiwix-tools tarball).

**Comment by nobicycle** [comment]
mgautierfr tip worked.
sudo -u kiwix /bin/bash -c "cd /opt/kiwix; /usr/lib/kiwix/bin/kiwix-manage /opt/kiwix/data/library/library.xml add /media/esata/MEDIA/System/Kiwix/data/content/wikipedia_en_for-schools_2013-01.zim"

Many thanks to him and kelson42
